src/configure/configure.ts

```diff
@@ -323,29 +342,7 @@ class Orchestrator {
         }
 
         try {
-            while (!this.inputs.sourceRepository.commitId) {
-                let commitOrDiscard = await vscode.window.showInformationMessage(
-                    utils.format(Messages.modifyAndCommitFile, Messages.commitAndPush, this.inputs.sourceRepository.branch, this.inputs.sourceRepository.remoteName),
-                    Messages.commitAndPush,
-                    Messages.discardPipeline);
-
-                if (commitOrDiscard && commitOrDiscard.toLowerCase() === Messages.commitAndPush.toLowerCase()) {
-                    await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: Messages.configuringPipelineAndDeployment }, async (progress) => {
-                        try {
-                            // handle when the branch is not upto date with remote branch and push fails
-                            this.inputs.sourceRepository.commitId = await this.localGitRepoHelper.commitAndPushPipelineFile(this.inputs.pipelineParameters.pipelineFilePath, this.inputs.sourceRepository, this.inputs.sourceRepository.repositoryProvider === RepositoryProvider.AzureRepos ? Messages.addAzurePipelinesYmlFile : Messages.addGitHubWorkflowYmlFile);
-                        }
-                        catch (error) {
-                            telemetryHelper.logError(Layer, TracePoints.CheckInPipelineFailure, error);
-                            vscode.window.showErrorMessage(utils.format(Messages.commitFailedErrorMessage, error.message));
-                        }
-                    });
-                }
-                else {
-                    telemetryHelper.setTelemetry(TelemetryKeys.PipelineDiscarded, 'true');
-                    throw new UserCancelledError(Messages.operationCancelled);
-                }
-            }
+            this.inputs.sourceRepository.commitId = await pipelineConfigurer.checkInPipelineFileToRepository(this.inputs, this.localGitRepoHelper);
```

Comment:
Why do you need to set commidId here? Why can't we set it along with other parameters in checkInPipelineFileToRepository function ?